### PR TITLE
fix: use pinned dependencies in gh actions

### PR DESCRIPTION
**Pull request by szuecs**
fix: use best practice from openssf and use pinned dependencies in gh actions

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -37,15 +37,15 @@ jobs:
 
     steps:
     - name: Checkout repository
-      uses: actions/checkout@v2
+      uses: actions/checkout@ee0669bd1cc54295c223e0bb666b733df41de1c5
```
Comment (AlexanderYastrebov) [inline .github/workflows/codeql-analysis.yml:40]: How did you obtain these hashes?

**Comment by AlexanderYastrebov** [comment]
:+1:

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (szuecs) [reply .github/workflows/codeql-analysis.yml:40]: https://github.com/actions/checkout/releases/tag/v2.7.0 and then click the small hash to get https://github.com/actions/checkout/commit/ee0669bd1cc54295c223e0bb666b733df41de1c5 , then copy the long sha

**Comment by szuecs** [comment]
:+1:

**Pull request merged**